### PR TITLE
add cross_entropy to nn/layer and nn/functional, test=develop

### PR DESCRIPTION
**Pull request by chajchaj**
### PR types
 New features 

### PR changes
 APIs 

### Describe
add cross_entropy to nn/layer and nn/functional

### CODE REVIEW

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -569,3 +569,132 @@ def mse_loss(input, label, reduction='mean', name=None):
...
+
+    If weight is not ``None``:
+
+    .. math::
+
+        loss_j =  \\text{weight[class]}(-\\text{input[class]} +
+        \\log\\left(\\sum_{i=0}^{K}\\exp(\\text{input}_i)\\right)), j = 1,..., K
+
+    Parameters:
+        input (Variable): Input tensor, the data type is float32, float64. Shape is
```
Comment (yaoxuefeng6) [inline python/paddle/nn/functional/loss.py]: Tensor

File: python/paddle/nn/functional/loss.py [hunk 2]
```diff
@@ -569,3 +569,132 @@ def mse_loss(input, label, reduction='mean', name=None):
...
+	    (N, C), where C is number of classes, and if shape is more than 2D, this
+	    is (N, C, D1, D2,..., Dk), k >= 1. 
+        label (Variable): Label tensor, the data type is int64. Shape is (N), where each
```
Comment (yaoxuefeng6) [inline python/paddle/nn/functional/loss.py]: Tensor

File: python/paddle/nn/functional/loss.py [hunk 3]
```diff
@@ -569,3 +569,132 @@ def mse_loss(input, label, reduction='mean', name=None):
...
+        \\log\\left(\\sum_{i=0}^{K}\\exp(\\text{input}_i)\\right)), j = 1,..., K
+
+    Parameters:
+        input (Variable): Input tensor, the data type is float32, float64. Shape is
+	    (N, C), where C is number of classes, and if shape is more than 2D, this
+	    is (N, C, D1, D2,..., Dk), k >= 1. 
+        label (Variable): Label tensor, the data type is int64. Shape is (N), where each 
+	    value is 0 <= label[i] <= C-1, and if shape is more than 2D, this is
+	    (N, D1, D2,..., Dk), k >= 1.
+        weight (Variable, optional): Weight tensor, a manual rescaling weight given
```
Comment (yaoxuefeng6) [inline python/paddle/nn/functional/loss.py]: Tensor

File: python/paddle/nn/functional/loss.py [hunk 4]
```diff
@@ -569,3 +569,132 @@ def mse_loss(input, label, reduction='mean', name=None):
...
+            If :attr:`size_average` is ``'sum'``, the reduced sum loss is returned.
+            If :attr:`reduction` is ``'none'``, the unreduced loss is returned.
+            Default is ``'mean'``.
+        ignore_index (int64, optional): Specifies a target value that is ignored
+            and does not contribute to the input gradient. Default is ``-100``.
+
+    Returns:
+        The tensor variable storing the cross_entropy_loss of input and label.
+
+    Return type: Variable.
```
Comment (yaoxuefeng6) [inline python/paddle/nn/functional/loss.py]: Tensor

File: python/paddle/nn/functional/loss.py [hunk 5]
```diff
@@ -22,7 +22,7 @@
 from ...fluid.layers.nn import _elementwise_op_in_dygraph
 from ...fluid.layers import bpr_loss  #DEFINE_ALIAS
 from ...fluid.layers import center_loss  #DEFINE_ALIAS
-from ...fluid.layers import cross_entropy  #DEFINE_ALIAS
+#from ...fluid.layers import cross_entropy  #DEFINE_ALIAS
```
Comment (willthefrog) [inline python/paddle/nn/functional/loss.py]: delete

File: python/paddle/nn/layer/loss.py [hunk 6]
```diff
@@ -117,7 +117,7 @@ class CrossEntropyLoss(fluid.dygraph.Layer):
                 print(output.numpy())
     """
 
-    def __init__(self, weight=None, reduction='mean', ignore_index=-100):
+    def __init__(self, weight=None, ignore_index=-100, reduction='mean'):
```
Comment (willthefrog) [inline python/paddle/nn/layer/loss.py:120]: use functional version in `forward()`?

**Review by willthefrog — APPROVED** [review]
LGTM

**Comment by jzhang533** [comment]
cross_entropy忘了加name参数吧？

**Pull request merged**